docs/usage.rst

```diff
@@ -159,6 +159,55 @@ Now we're going to get the base URL and some image links::
      u'image4_thumb.jpg',
      u'image5_thumb.jpg']
 
+.. _topics-selectors-css-extensions:
+
+Extensions to CSS Selectors
+---------------------------
+
+Per W3C standards, `CSS selectors`_ do not support selecting text nodes
+or attribute values.
+But selecting these is so essential in a web scraping context
+that Parsel implements a couple of **non-standard pseudo-elements**:
+
+* to select text nodes, use ``::text``
+* to select attribute values, use ``::attr(name)`` where *name* is the
+  name of the attribute that you want the value of
+
+.. warning::
+    These pseudo-elements are Scrapy-/Parsel-specific.
+    They will most probably not work with other libraries like `lxml`_ or `PyQuery`_.
+
+
+Examples:
+
+* ``title::text`` selects children text nodes of a descendant ``<title>`` element::
+
+    >>> selector.css('title::text')
+    [<Selector xpath=u'descendant-or-self::title/text()' data=u'Example website'>]
+
+
+* ``*::text`` selects all descendant text nodes of the current selector context::
+
+    >>> selector.css('*::text')
+    [<Selector xpath=u'descendant-or-self::text()' data=u'\n '>,...
+
+* ``a::attr(href)`` selects the *href* attribute value of descendant links::
+
+    >>> selector.css('a::attr(href)').extract()
+    [u'image1.html',
+     u'image2.html',
+     u'image3.html',
+     u'image4.html',
+     u'image5.html']
```

Comment:
A nice example! Could we have something like this for ::text? E.g. in the first example most users will want `selector.css('title::text').extract_first()`